### ISSUE TITLE
Rule Change: `no-useless-escape` support for `pattern` usage

### ISSUE THREAD

**Issue by benglewis** [eslint/eslint#18839]
### What rule do you want to change?

no-useless-escape

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const myRegex = /[\/]/
```

### What does the rule currently do for this code?

Unnecessary escape character: \/.eslint[no-useless-escape](https://eslint.org/docs/latest/rules/no-useless-escape)

### What will the rule do after it's changed?

It should be possible to turn off this warning, since it the escape character before the `/` in the square brackets is necessary for the `pattern` argument for `input` HTML elements (It returns this error when removing the escape character "Pattern attribute value [/] is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /[/]/v: Invalid character in character class")

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @benglewis, thanks for the issue!

> It should be possible to turn off this warning, since it the escape character before the `/` in the square brackets is necessary for the `pattern` argument for `input` HTML elements (It returns this error when removing the escape character "Pattern attribute value [/] is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /[/]/v: Invalid character in character class")

Can you please provide a repro repo that demonstrates this behavior? Note that the evaluated regular expression has the `v` flag (`Invalid regular expression: /[/]/v`). ESLint by default doesn't parse HTML, so if you're getting this error in an HTML file then you're probably using a plugin, and the plugin doesn't seem to add the `v` flag to these regular expressions. Note that the `no-useless-escape` rule doesn't report escaped slashes when the regular expression has the `v` flag:

```
/*eslint no-useless-escape: "error"*/

const myRegex = /[\/]/; // error

const myRegexWithVFlag = /[\/]/v; // ok
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdXNlbGVzcy1lc2NhcGU6IFwiZXJyb3JcIiovXG5cbmNvbnN0IG15UmVnZXggPSAvW1xcL10vOyAvLyBlcnJvclxuXG5jb25zdCBteVJlZ2V4V2l0aFZGbGFnID0gL1tcXC9dL3Y7IC8vIG9rXG5cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)

**Comment by mdjermanovic** [comment]
No response so closing.
